### PR TITLE
`InfileError` improvements

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Addressed review issues.

- Added a docstring with an example usage.
- Added a comment inside the conditional that appends to the supplied message.
- Removed "majic number".
- Streamlined a bit.

I have not switched from placeholders to f-strings, but only because I want to move on to the next effort.  If you would like to still see that change, I'll create an issue.

## Affected Issues/Pull Requests

- Resolves no issue
- Merges into PR #901

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/utils/exceptions.py [hunk 1]
```diff
@@ -17,26 +17,80 @@
...
+        ValueError: If the length of the order list doesn't match the `%s` occurrences in the message.
+
+    Returns:
+        Instance
+    """
+
     def __init__(
         self, message, rownum=None, sheet=None, file=None, column=None, order=None
     ):
+        location_args = ["rownum", "column", "file", "sheet"]
```
Comment (hepcat72) [inline DataRepo/utils/exceptions.py:70]: C. This is how I removed the magic number.

File: DataRepo/utils/exceptions.py [hunk 2]
```diff
@@ -56,14 +110,13 @@ def __init__(
...
-            if "loc" not in order and len(order) != 4:
-                insertions.append(loc)
+            vdict["loc"] = loc
+            self.loc = loc
+            insertions = [vdict[k] for k in order]
             message = message % tuple(insertions)
         else:
             message = message % loc
         super().__init__(message)
-        self.loc = loc
```
Comment (hepcat72) [inline DataRepo/utils/exceptions.py:66]: C. Moved up.

File: DataRepo/utils/exceptions.py [hunk 3]
```diff
@@ -17,26 +17,80 @@
...
+            # messages in a uniform way.  It inserts that information where `%s` occurs in the supplied message.
+            # Instead of making `%s` (in the simplest case) required, and raise an exception, the `%s` is just appended.
             message += "  Location: %s."
+
         if order is not None:
-            if "loc" not in order and len(order) != 4:
-                if message.count("%s") != len(order) + 1:
+            if "loc" not in order and len(order) != len(location_args):
+                order.append("loc")
+                if message.count("%s") != len(order):
```
Comment (hepcat72) [inline DataRepo/utils/exceptions.py:89]: C. Got rid of the `+ 1` by moving the append up.

File: DataRepo/utils/exceptions.py [hunk 4]
```diff
@@ -56,14 +110,13 @@ def __init__(
             loc = generate_file_location_string(
                 rownum=rownum, sheet=sheet, file=file, column=column
             )
-            insertions = [vdict[k] if k != "loc" else loc for k in order]
-            if "loc" not in order and len(order) != 4:
-                insertions.append(loc)
+            vdict["loc"] = loc
+            self.loc = loc
+            insertions = [vdict[k] for k in order]
```
Comment (hepcat72) [inline DataRepo/utils/exceptions.py:115]: C. I was able to simplify this.  I realized that `loc` was already appended from above and that if I just added a `loc` key to the `vdict`, I didn't need to use a conditional in the list comprehension.

**Review by lparsons — APPROVED** [review]
Looks great, it's always good to have nice docstrings, esp. for base classes.

**Pull request merged**